### ISSUE TITLE
Sending email with both body and attachment - body text added as attachment

### ISSUE THREAD

**Issue by stankut** [putyourlightson/craft-sendgrid#12]
### Describe the bug
I have troubles using this plugin for sending email that would have text body + attachment.
When an attachment is added to the email, the normal email body text automatically becomes another attachment.

If however there is no attachment added to the email - the email has the normal text body.

### To reproduce
Steps to reproduce the behaviour:

1. Select SendGrid as email transport
2. Use the following code to send email:
```
$pdfFilePath = 'path/to/pdf';

$result = Craft::$app
->getMailer()
->compose()
->setTo($recipient)
->setSubject($this->getSubjectFromSettings())
->setTextBody('my body')
->attach($pdfFilePath)
->send();
```

2. Email is delivered with **PDF attachment**, but body is **NOT** visible - it is added as another attachment to the email. 
3. Change the email transport type to default one (sendmail)
4. Repeat email sending process
5. Email is delivered with **PDF attachment**, body added to the email in a **regular way**

### Expected behaviour
Email with body and attachment shall be delivered via SendGrid so that body is added regularly, and not as an another attachment.

### Screenshots
https://ibb.co/NTGprhM

### Versions
- Plugin version: 1.2.2
- Craft version: 3.7.13

**Comment by bencroker** [comment]
If you set `Craft::$app->getMailer()->useFileTransport = true;` before sending the email, the result will be output to `storage/mail/`. Do you also see this when you open the `.eml` file?

<img width="599" alt="Screenshot 2021-10-23 at 20 35 22" src="https://user-images.githubusercontent.com/57572400/138567702-6776a711-0f3e-4890-a4d2-d0085da07ed5.png">

**Comment by stankut** [comment]
@bencroker this is what i am getting:

![Screenshot 2021-10-25 at 09 35 42](https://user-images.githubusercontent.com/1537807/138646034-1a28de4c-3cb3-4cef-a862-66c3c9ef9c96.png)

**Comment by bencroker** [comment]
What if you try setting the body as follows?

```twig
->setHtmlBody('my body')
```

**Comment by stankut** [comment]
@bencroker then it looks like this, but the body is still treated as an attachment

![Screenshot 2021-10-25 at 10 28 34](https://user-images.githubusercontent.com/1537807/138652836-a4ffbcad-a511-4672-abda-891ff5c14805.png)

**Comment by stankut** [comment]
@bencroker 
correction: both "Content-Type: text/html; charset=utf-8" and "Content-Type: plain/text; charset=utf-8"

are working OK, when it's first saved into a .eml file and opened with an email client.

However if the email is sent directly via SendGrid (useFileTransport is set to FALSE) - then the headers looks differently.
![Screenshot_2021-10-25_at_10_38_58](https://user-images.githubusercontent.com/1537807/138654319-67d7068d-1993-41ef-a707-456e98e592af.jpg)

**Comment by bencroker** [comment]
So am I correct in thinking that the email body doesn't even exist there?

**Comment by stankut** [comment]
when sending via SendGrid, email body is empty, and the text is added as another attachment.

![Screenshot 2021-10-25 at 10 53 21](https://user-images.githubusercontent.com/1537807/138656398-658b8904-244e-40b7-892a-cc13dfe935d9.png)

![Screenshot 2021-10-25 at 10 54 00](https://user-images.githubusercontent.com/1537807/138656408-cdec35fc-96f4-4b0c-b192-613e247ace95.png)

**Comment by bencroker** [comment]
Got that, but how does the body appear in the message source code? I don't see it in your screenshot above.

**Comment by bencroker** [comment]
Can you please also forward the email to info@putyourlightson.com so I can take a look?

**Comment by stankut** [comment]
@bencroker i emailed you the information
thanks in advance

**Comment by bencroker** [comment]
Interestingly, when I open the email in Apple Mail, all seems well, so perhaps this has to do with your email client?

<img width="849" alt="Screenshot 2021-10-25 at 12 51 16" src="https://user-images.githubusercontent.com/57572400/138683206-c5e6665e-eec2-48a1-93f7-0ea09422cec3.png">

In saying that, I see that the `Content-Type` of the body is set to `multipart/mixed`, which may be the cause of the issue. Can you check the value of `Content-Type` of the body in an email that works as expected, with body text and an attachment?

<img width="703" alt="Screenshot 2021-10-25 at 12 52 41" src="https://user-images.githubusercontent.com/57572400/138683566-6e8a304f-9cea-4bda-b2bc-39a581fae301.png">

**Comment by stankut** [comment]
that's not a single email client.
I tested with Gmail (web version) and Thunderbird - text is not visible.

Perhaps Apple client is treating it differently?

anyways, i emailed you the email example that opens fine with text (it was sent NOT using SendGrid plugin)

**Comment by bencroker** [comment]
Great, so in the version that works the body follows `Content-Type: text/html`. I'll need to do some digging into why it is missing when using Sendgrid.

**Comment by stankut** [comment]
hi @bencroker 
did you have any progress with this issue?

**Comment by bencroker** [comment]
I believe this should be fixed in https://github.com/putyourlightson/craft-sendgrid/commit/a042cb8b842c15033350c416a5ffeec2e9fc0790, can you test and let me know?

**Comment by stankut** [comment]
@bencroker hi, it works perfectly, thank you.
Will you make it as a release?

**Comment by bencroker** [comment]
Released in version 1.2.3.
